monai/losses/aucm_loss.py

```diff
@@ -0,0 +1,172 @@
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from typing import Callable, List, Optional, Union
+
+import torch
+from torch.nn.modules.loss import _Loss
+
+from monai.networks import one_hot
+from monai.utils import LossReduction
+
+
+class AUCMLoss(_Loss):
+    """
+    AUCM Loss with squared-hinge function: a novel loss function to directly optimize AUROC
+
+    The original paper: Large-scale Robust Deep AUC Maximization: A
+    New Surrogate Loss and Empirical Studies on Medical Image Classification
+    https://arxiv.org/abs/2012.03173
+    """
+
+    def __init__(
+        self,
+        margin: float = 1.0,
+        pos_weight: List[float] = None,
+        num_classes: int = 2,
+        to_onehot_y: bool = False,
+        sigmoid: bool = False,
+        softmax: bool = False,
+        other_act: Optional[Callable] = None,
+        reduction: Union[LossReduction, str] = LossReduction.MEAN,
+    ):
+        """
+        Args:
+            margin: margin term for AUCM loss, default 1.0, range [0, 1]
+            imratio: imbalance ratio, ratio of positive and negative samples, default None
+            num_classes: number of classes, default 2
+            to_onehot_y: whether to convert `y` into the one-hot format. Defaults to False.
+            sigmoid: if True, apply a sigmoid function to the prediction.
+            softmax: if True, apply a softmax function to the prediction.
+            other_act: if don't want to use `sigmoid` or `softmax`, use other callable function to execute
+                other activation layers, Defaults to ``None``. for example:
+                `other_act = torch.tanh`.
+            reduction: {``"none"``, ``"mean"``, ``"sum"``}
+                Specifies the reduction to apply to the output. Defaults to ``"mean"``.
+
+                - ``"none"``: no reduction will be applied.
+                - ``"mean"``: the sum of the output will be divided by the number of elements in the output.
+                - ``"sum"``: the output will be summed.
+            gpu: if True, use GPU to calculate. Defaults to False.
+
+
+        Raises:
+            ValueError: margin must be in [0, 1]
+            TypeError: When ``other_act`` is not an ``Optional[Callable]``.
+            ValueError: and When more than 1 of [``sigmoid=True``, ``softmax=True``, ``other_act is not None``].
+                Incompatible values.
+        """
+        super().__init__(reduction=LossReduction(reduction).value)
+
+        if margin < 0 or margin > 1:
+            raise ValueError("imratio must be between 0 and 1")
```

Comment:
Maybe `raise ValueError("margin must be between 0 and 1")`?
But I didn't find where use this `margin` arg, does it need?